src/OpenTelemetry/Logs/Builder/LoggerProviderBuilderSdk.cs

```diff
@@ -0,0 +1,133 @@
+    }
+
+    public List<InstrumentationRegistration> Instrumentation { get; } = new();
+
+    public ResourceBuilder? ResourceBuilder { get; private set; }
+
+    public LoggerProvider? Provider => this.loggerProvider;
+
+    public List<BaseProcessor<Activity>> Processors { get; } = new();
+
+    public void RegisterProvider(LoggerProviderSdk loggerProvider)
+    {
+        Debug.Assert(loggerProvider != null, "loggerProvider was null");
+
+        if (this.loggerProvider != null)
+        {
+            throw new NotSupportedException("LoggerProvider cannot be accessed while build is executing.");
+        }
+
+        this.loggerProvider = loggerProvider;
+    }
+
+    public override LoggerProviderBuilder AddInstrumentation<TInstrumentation>(
+        Func<TInstrumentation> instrumentationFactory)
+    {
+        Debug.Assert(instrumentationFactory != null, "instrumentationFactory was null");
+
+        this.Instrumentation.Add(
+            new InstrumentationRegistration(
+                typeof(TInstrumentation).Name,
+                typeof(TInstrumentation).Assembly.GetName().Version?.ToString() ?? DefaultInstrumentationVersion,
+                instrumentationFactory!()!));
+
+        return this;
+    }
+
+    public LoggerProviderBuilder ConfigureResource(Action<ResourceBuilder> configure)
+    {
+        Debug.Assert(configure != null, "configure was null");
+
+        var resourceBuilder = this.ResourceBuilder ??= ResourceBuilder.CreateDefault();
+
+        configure!(resourceBuilder);
+
+        return this;
+    }
+
+    public LoggerProviderBuilder SetResourceBuilder(ResourceBuilder resourceBuilder)
+    {
+        Debug.Assert(resourceBuilder != null, "resourceBuilder was null");
+
+        this.ResourceBuilder = resourceBuilder;
+
+        return this;
+    }
+
+    public LoggerProviderBuilder AddProcessor(BaseProcessor<Activity> processor)
+    {
+        Debug.Assert(processor != null, "processor was null");
+
+        this.Processors.Add(processor!);
+
+        return this;
+    }
+
+    public LoggerProviderBuilder ConfigureBuilder(Action<IServiceProvider, LoggerProviderBuilder> configure)
+    {
+        Debug.Assert(configure != null, "configure was null");
+
+        configure!(this.serviceProvider, this);
+
+        return this;
+    }
+
+    public LoggerProviderBuilder ConfigureServices(Action<IServiceCollection> configure)
+    {
+        throw new NotSupportedException("Services cannot be configured after ServiceProvider has been created.");
+    }
+
+    LoggerProviderBuilder IDeferredLoggerProviderBuilder.Configure(Action<IServiceProvider, LoggerProviderBuilder> configure)
+        => this.ConfigureBuilder(configure);
+
+    internal readonly struct InstrumentationRegistration
+    {
+        public readonly string Name;
+        public readonly string Version;
+        public readonly object Instance;
+
+        internal InstrumentationRegistration(string name, string version, object instance)
```

Comment:
Not related to this PR but could we share this struct among the different provider SDKs?

Reply:
Totally. Or just remove it, I don't think we use this data for anything?